spec/view_code_pr_checker_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::ViewCodePRChecker do
+    it 'should be a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    shared_examples 'view code without screenshots' do |modified_files|
+      before do
+        allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+      end
+
+      it 'warns when a PR with view code changes does not have screenshots' do
+        allow(@plugin.github).to receive(:pr_body).and_return('PR Body')
+
+        @plugin.view_changes_need_screenshots
+
+        expect(@dangerfile.status_report[:warnings].count).to eq 1
+      end
+
+      it 'does nothing when a PR with view code changes has screenshots' do
```

Comment:
You put that shared example about "has view code and also **has** screenshots" inside the `shared_examples` that is named "view code **without** screenshots". This seems contradictory.

The easiest fix for that is probably to rename the parent `shared_examples` to just `'PR has view code'`.